api docs: Convert remaining JavaScript examples to use the tested system.

akashaviator opened the pull request:
Fixes #15496.

akashaviator commented:
@timabbott Can you have a look ?

timabbott commented:
Can you post notes on any examples where the content changed?

@aero31aero it'd be great if you can review since you're more familiar with the JS bindings than I am.

akashaviator commented:
I have changed the content slightly in the
* `send_message` (I've mentioned in one of the commit messages)
* `get_messages` (there was a bug in the original example, so I changed it and made it almost equal to its python counterpart)
* `update_message_flags` ( made it similar to its python counterpart as it was more suited to document with the current system which we certainly has some scope of improvement.)

timabbott commented:
I feel like this example ends up being a bit weird -- I think we want the "create_queue" part to be outside what we show in the documentation here.


![image](https://user-images.githubusercontent.com/2746074/85466637-1100b100-b55f-11ea-8b4e-ad2efcaa59c0.png)

timabbott commented:
Overall this is great; the things I'd complain about were issues with the old/untested examples or are otherwise orthogonal to the migration to actually test the logic.

I merged all but the last commit to avoid closing the issue, because you're also missing `get_events.md`, after making a small change to delete unnecessary parameters in the `update_message_flags` example.  

Let me know when you have this updated to remove the last old-style example :)

aero31aero commented:
I'm taking a look at this locally now. This looks good at a glance!

timabbott commented:
Yeah, several examples are weird, but they were often weird with the previous version, so I think that further improvements is something we should just do on top of this work.

timabbott commented:
@akashaviator it looks like you fixed this but neglected to comment that you had?

timabbott commented on zerver/openapi/javascript_examples.js:
```diff
@@ -287,6 +286,7 @@ add_example('update_message_flags', '/messages/flags:post', 200, async (client)
     // Add the "read" flag to the messages with IDs in "message_ids"
     const addflag = {
         messages: message_ids,
+        op: 'add',
```
These `op` changes are unnecessary.

timabbott replied:
(And I think this whole commit is wrong and should be dropped; we already merged the correct version).  

You just need to replace this with a commit fixing the last unmigrated file in `templates/zerver/api/get-events.md`.

akashaviator replied:
@timabbott Actually, I just rebased the branch and pushed (and didn't look too closely what was getting changed.). I was yet to start on moving the last example from `get-events.md`.

akashaviator commented:
@timabbott @orientor Seems like the "queue_id" parameter is not documented in the responses for GET /events in zulip.yaml. Should I add it ?
Also, the Python example for GET /events resides in the `get_events.md`(that was the reason I didn't move the JS example earlier). Should I move it too ?

akashaviator commented:
The JS example in get-events.md takes a lot of time to return the result.
```js
    const eventParams = {
        queue_id: queue_id,
        last_event_id: -1,
        dont_block: false,
    };

    return await client.events.retrieve(eventParams);
```
Having `dont_block` set to `true` gives response instantly but without `queue_id`. Should we keep it this way ?

timabbott commented:
Good question re: that example.  The thing is, what we want to document is to not use `dont_block`, because that's a weird choice, but we also don't want to waste a minute here.  I think we can hack around this with the following question:
* Call register as shown in the test
* Call `send_message` in JS code, outside the blocks of code we show users.
* Call `get_events` without `dont_block`, which will return the message we sent.

Or something like that.

akashaviator commented:
@timabbott Can you have a look ?

timabbott commented:
Merged after fixing the merge conflicts and tweaking some whitespace, thanks @akashaviator!
